### ISSUE TITLE
Integrate structs in npc_battle.h and battle_structs.h

### ISSUE THREAD

**Issue by emilybae1705** [uchicago-cs/chiventure#1283]
As laid out in the [NPC Battle Integration design document](https://github.com/uchicago-cs/chiventure/wiki/NPC-~-NPC-Battle-Integration-Design-Document), the `npc_t `and `npc_battle_t` structs in `npc.h` need modification in terms of having a specific battle item list in `npc_battle_t` extracted from the inventory list in `npc_t` as well as incorporating npc class type. These changes can be made by referencing `combatant_t` in `battle_common.h` and `stat_t `and `battle_item_t `in `battle_structs.h`.

**Comment by srydbergcox** [comment]
In looking at this change, clearly, the changing of the struct will be relatively straight forward. But, I am concerned about the changes that this will require given that changing this struct means that the `npc_battle_init` & `npc_battle_new` functions will have to change their parameters which will cause compiler errors for everywhere in the code base that these functions are called. So, I need to assess how often these functions are called.

**Comment by srydbergcox** [comment]
I have added the `class_t` struct to the `npc_battle_t` and made the necessary changes to the `npc_battle_init` & `npc_battle_new` functions. It now breaks some tests from the battle team which are not expecting this class_t parameter and I am waiting on senior dev clarification to proceed.

**Comment by srydbergcox** [comment]
Now, I have fixed all test cases that broke with the update to `npc_battle_t` struct. Chiventure compiles and I am now moving on to adding the npc's inventory to the `npc_battle_t` struct.

**Comment by srydbergcox** [comment]
After looking at incorporating the npc's inventory into the `npc_battle_t` struct, it seems that there is a type conflict as npc inventory is stored in `item_hash_t` while the battle team/structs are looking for `battle_item_t`. i think this may warrant a separate issue. I will bring this up at our team meeting this afternoon and update this issue.

**Comment by srydbergcox** [comment]
After our team meeting on Monday, we have decided to simply incorporate the `battle_item_t` type into the `npc_battle_t` struct as the battle team said that they would take on the conversion from `item_hash_t` to `battle_item_t`.

**Comment by srydbergcox** [comment]
Sam's closing statement: 
My PR for this issue is now complete. In this issue, we identified and implemented changes to the `npc_battle_t` struct to increase the variety of battles. Specifically, I added the `class_type_t` and `battle_item_t` structs. This also required updates to the `npc_battle_new`, `npc_battle_init`, and `add_battle_to_npc` functions to add these new structs in as parameters. Then, this required me to update each time that these functions were called in Chiventure. Overall, these changes make it so that we can have a variety of opponents for players to fight with as we can modify an NPC's stats, moves, etc. based on their player class in the future. Also, adding in battle items increases the diversity of moves that NPCs can have available to them.

**Comment by emilybae1705** [comment]
In continuation of Sam's closing statement, my PR for this issue has been approved and merged into dev branch. My PR removed `surrender_level` and `health` from the `npc_battle_t` struct as `surrender_level` has been added to `stat_t` and `health` has been renamed as `hp` in the `stat_t` struct. I also replaced bool `will_fight` in `npc_t` to `hostility_t` so that we can make `npc_battle_t` and `npc_t` more compatible. That way, it would be easier to implement a hostility_level conversion function that changes an NPC's hostility_level once a player provokes them in dialogue (see #1514 for more details). These changes required multiple changes to both npc and battle modules as well as `test_room.c` which used the `add_battle_to_npc()` function. These changes now make for smoother integration between battle structs and NPC structs and improves battle functions that start a battle between a combatant player and an NPC.
